File: app/src/mainJava/java/com.playground/data/database/NoteEntry.java
package com.playground.data.database;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;

import org.jetbrains.annotations.NotNull;

import java.sql.Date;

/**
 * NoteEntry presentation
 */
@Entity(tableName = "note")
public class NoteEntry {
    public static final String DATA = "note_data";
    public static final int DEFAULT_ID = 0;

    @PrimaryKey(autoGenerate = true)
    private int id;
    private String title;
    private String description;
    private int priority;


    @ColumnInfo(name = "updated_at")
    private Date updatedAt;


    public NoteEntry() {
    }

    @Ignore
    public NoteEntry(int noteId, @NotNull String title, @NotNull String description) {
        this.id = noteId;
        this.title = title;
        this.description = description;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getPriority() {
        return priority;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    @Override
    public String toString() {
        return "NoteEntry{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", priority=" + priority +
                ", updatedAt=" + updatedAt +
                '}';
    }
}

File: app/src/mainKotlin/kotlin/com.playground/entities/NoteEntry.kt
package com.playground.entities

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import java.sql.Date


/**
 * NoteEntry presentation
 */
@Entity(tableName = "note")
data class NoteEntry(@PrimaryKey(autoGenerate = true) val id: Long,
                     val title: String,
                     val description: String,
                     val priority: Int,
                     @ColumnInfo(name = "updated_at") val updatedAt: Date?) {

    override fun toString(): String {
        return "NoteEntry(id=$id, title='$title', description='$description', priority=$priority, updatedAt=$updatedAt)"
    }

    companion object {
        const val DATA = "note_data"
        const val DEFAULT_ID: Long = 0
    }


}